proto/destination.proto

```diff
@@ -86,6 +86,7 @@ message WeightedAddr {
   uint32 weight = 3;
   map<string, string> metric_labels = 4;
   TlsIdentity tls_identity = 5;
+  bool linkerd_managed = 6;
```

Comment:
1) I'd prefer not to use the word _linkerd_ in the API, having had to rename the project once already.

2) What do you think about generalizing this as a `ProtocolHint` oneof -- the default should be an empty Unknown, and we'd include an `H2` (or H2C?) variant. -- A oneof rather than an enum so that proto-specific metadata can be added.

My general intent here is to make the intent/behavior explicit rather than implicit. An implementor of this API would have a hard time being sure about what behavior is influenced by a _managed_ bool; a protocol hint field is pretty hard to misunderstand.

Reply:
Sure, I can do that. Though, the name `ProtocolHint` to *me* suggests that the service itself speaks a certain protocol, not that there is a linkerd proxy in place to handle it... I can't think of a better name, though.